### PR TITLE
Add a new option for output name of the total scattering pdf 

### LINKED ISSUE #34750: Add extra suffix to PDF output workspace to contain PDF type
When generating multiple pdf's from a single focussed dataset (eg different pdf types) it would be useful if the output workspace didn't have the same name. At the moment I have to manually rename the output workspace to keep the different outputs.

So either of these would be useful (or both):
a) adding PDF type to the output workspace name
b) allow the call to create_total_scattering_pdf to take an output name parameter.

### PR DESCRIPTION
**Pull request by MohamedAlmaki**
### Description of work
This pull request introduces a new parameter, "pdf_output_name," to the "create_total_scattering_pdf" function. With this new parameter, users can customise the output file name for the total scattering PDF. Additionally, the default file name for the PDF has been updated to include the PDF type as a suffix.

Fixes #34750. 

### To test:
1- Run the following script:

```python
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

from isis_powder import polaris, SampleDetails

config_file_path = r"C:\Users\xhg73778\Downloads\POLARISPowderDiffractionFolderSmall\polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="test", mode="PDF")

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si', packing_fraction=0.6)
polaris.set_sample_details(sample=sample_details)

polaris.create_vanadium(first_cycle_run_no="125706", multiple_scattering=True, 
                         do_absorb_corrections=True, mayers_mult_scat_events=1)                  

polaris.focus(run_number="125754", input_mode='Summed', do_absorb_corrections=False,
              do_van_normalisation=False, van_normalisation_method="Relative",
              keep_raw_workspace=False)  # input_mode='individual', input_mode='Summed'

q_lim = [[2.5, 3, 4, 6, 7], [3.5, 5, 7, 11, 40]]
polaris.create_total_scattering_pdf(run_number='125754', merge_banks=True, q_lims=q_lim, pdf_type='g(r)')

```
2- Check the output name of the pdf in the workspaces window and it should be '125754_pdf_g(r)'
3- Run the same above script and the parameter `pdf_output_name='test_output'` to `create_total_scattering_pdf`
4- Check the output name of the pdf in the workspaces window and it should be 'test_output'

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Testing/SystemTests/tests/framework/ISIS_PowderPolarisTest.py [hunk 1]
```diff
@@ -447,6 +447,21 @@ def validate(self):
...
+
+    def runTest(self):
+        setup_mantid_paths()
+        # Load Focused ws
+        mantid.LoadNexus(Filename=total_scattering_input_file, OutputWorkspace="98533-ResultTOF")
+        q_lims = np.array([2.5, 3, 4, 6, 7, 3.5, 5, 7, 11, 40]).reshape((2, 5))
+        self.pdf_output = run_total_scattering("98533", True, q_lims=q_lims, pdf_type="g(r)", pdf_output_name="test_pdf_output")
+
+    def validate(self):
+        self.assertEqual(self.pdf_output.name(), "test_pdf_output")
```
Comment (RichardWaiteSTFC) [inline Testing/SystemTests/tests/framework/ISIS_PowderPolarisTest.py:464]: Do we want to add a test for the default name as well or do you think that's overkill?

File: docs/source/techniques/ISISPowder-Polaris-v1.rst [hunk 2]
```diff
@@ -223,6 +223,7 @@ The output PDF can be customized with the following parameters:
 - By calling with `debug=True` which will retain the intermediate self scattering correction workspace.
 - By calling with `placzek_order` the Placzek correction order can be specified, with the option of 1 or 2 (defaults to 1).
 - By calling with `sample_temp` the user can override the sample temperature provided in the logs. It defaults to using values from the logs if available.
+- By calling with pdf_output_name, the name of the output PDF will be set to the user-provided name. If not specified, the output will be the run number suffixed with the PDF type.
```
Comment (RichardWaiteSTFC) [inline docs/source/techniques/ISISPowder-Polaris-v1.rst:226]: Think pdf_output_name should be in code format `pdf_output_name`

**Review by RichardWaiteSTFC — CHANGES_REQUESTED** [review]
Code works as expected, nice clean change, just a couple of really minor things!

File: Testing/SystemTests/tests/framework/ISIS_PowderPolarisTest.py (hunk 1 above)
Comment (MohamedAlmaki) [reply Testing/SystemTests/tests/framework/ISIS_PowderPolarisTest.py:464]: Hmm, it is a good idea, but I did that implicitly by adding line 450 in the other test. Do you think it is better to separate it into its own unit test?

Comment (RichardWaiteSTFC) [reply Testing/SystemTests/tests/framework/ISIS_PowderPolarisTest.py:464]: Ah yes I see, sorry about that! Looks like both cases are covered so I'm happy!

**Review by RichardWaiteSTFC — APPROVED** [review]
Looks great thanks!

**Pull request merged**